.github/workflows/build.yml

```diff
@@ -90,19 +90,28 @@ jobs:
           ruby-version: 2.6.5
       - uses: dart-lang/setup-dart@v0.3
         with:
-          sdk: stable
+          sdk: 2.10.5
       - name: Install node dependencies
         run: npm install
       - name: Install dart dependencies
         run: dart pub get
       - name: Select flutter
-        if: matrix.sdk != 'submodule-commit'
+        if: matrix.sdk != 'submodule-commit' && matrix.sdk != 'stable'
```

Comment:
I did add some comments, I hope that helps.

Though probably there is a better way. Flutter's tooling relies on clues in the git checkout to figure out which version it is and on what branch it is. GitHub Actions does a shallow, single-branch checkout which doesn't have all the metadata that is needed to switch Flutter channels properly. Normally, we'd have to fetch the entire repo. This code tries to preserve the shallow checkout and checkout only what's needed. Perhaps this could be simpler if we removed the flutter submodule, and then we could directly checkout the right branch instead.

On Travis, we did roughly this:
* Checkout stable (full checkout that fetches everything)
* Run flutter doctor to get the flutter tool.
* Use the flutter tool to switch channels (which internally does similar things as these git commands).
* Then, when flutter was used somewhere, it would fetch additional artifacts for the selected release channel.

Perhaps there's a better way to get a specific Flutter version, but with the tight integration with a git checkout I don't know any. For the Dart SDK, it's just "forge a URL to the zip file, then download and unpack". It would be possible to go back to the Travis approach, but that is rather inefficient.